runtime/lua/vim/lsp/buf.lua

```diff
@@ -258,10 +258,72 @@ function M.implementation(opts)
   get_locations(ms.textDocument_implementation, opts)
 end
 
+local sig_help_ns = api.nvim_create_namespace('vim_lsp_signature_help')
+
+--- @class vim.lsp.buf.signature_help.Opts : vim.lsp.util.open_floating_preview.Opts
+--- @field silent? boolean
+
+-- TODO(lewis6991): support multiple clients
 --- Displays signature information about the symbol under the cursor in a
 --- floating window.
-function M.signature_help()
-  lsp.buf_request(0, ms.textDocument_signatureHelp, client_positional_params())
+--- @param config? vim.lsp.buf.signature_help.Opts
+function M.signature_help(config)
+  local method = ms.textDocument_signatureHelp
+
+  config = config or {}
+  config.focus_id = method
+
+  lsp.buf_request(0, method, client_positional_params(), function(err, result, ctx)
```

Comment:
I assume proper merging of the results will be a follow up?

Reply:
Yeah, out of scope for this PR